drivers/sensor/b57861s0103a039/b57861s0103a039.c

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT epcos_b57861s0103a039_spi
```

Comment:
The file name and very long compat are pretty cryptic, can we abbreviate this a bit to epcos_b57861s.c with perhaps some DT attributes to help clarify which variant of the part family it is. E.g. the 100k/10k/etc part. Some parts will likely be exactly the same but qualified for auto and such.